Bottom: b6947824e1c60b1d04e12d07e4b21e2f725bf32d
Top:    3bf4f4d7aeb48afe4ac9139f4eafbcfbfb8601cc
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-22 04:50:10 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index f1f2ecd..6632e75 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -227,6 +227,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
